lib/Migration/Version5000Date20211025124248.php

```diff
@@ -0,0 +1,218 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OCA\User_SAML\Migration;
+
+use Closure;
+use OCP\DB\ISchemaWrapper;
+use OCP\DB\QueryBuilder\IQueryBuilder;
+use OCP\DB\Types;
+use OCP\IDBConnection;
+use OCP\Migration\IOutput;
+use OCP\Migration\SimpleMigrationStep;
+
+/**
+ * Auto-generated migration step: Please modify to your needs!
+ */
+class Version5000Date20211025124248 extends SimpleMigrationStep {
+
+	private const IDP_CONFIG_KEYS = [
```

Comment:
dejavu from SAMLSettings, or are we duplicating these on purpose to not class-load the other class before running the migration ? (please answer in a PHPDoc that clarifies if applicable)

Reply:
The two lists are not exactly the same. lib/SAMLSettings.php contains more entries. I'm not sure if this is expected or not. @blizzz

Reply:
It  is sort of duplicate for we cannot rely on new classes on migration as old ones may have been autoloaded. Also, when new keys are added to SAMLSettings, they are not required in the Migration class for they cannot have been added in the past. Therefore this snapshot is also reasonable.

@CarlSchwan  thanks for pointing out the differences! They should be there indeed. 

There is also an existing issue with security-sloWebServerDecode which is IdP-sensitive but was always read from the first config.  So, tasks:

* [ ] fix security-sloWebServerDecode – set globally, but IdP sensitive
* [ ] migrate sp-x509cert, sp-name-id-format, sp-privateKey, security-sloWebServerDecode
* [ ] evaluate reading of sp-name-id-format in AdminSettings getForm (before, it took the first from config).